lib/src/core_matchers.dart

```diff
@@ -625,14 +629,24 @@ class CustomMatcher extends Matcher {
 
   Description describeMismatch(
       item, Description mismatchDescription, Map matchState, bool verbose) {
+    if (matchState['exception'] != null) {
+      mismatchDescription.add('threw ').addDescriptionOf(matchState['exception']);
+      if (verbose) {
+        mismatchDescription.add(' at ').add(matchState['stack'].toString());
```

Comment:
Add a newline before the stack trace; "at" is probably not necessary.

To be maximally usable from within tests, this should include the [stack chain](https://github.com/dart-lang/stack_trace/#stack-chains). You'll have to add `stack_trace` as a dependency.